File: src/main/scala/top/theonly/spark/sca/sql/textfile/SparkReadCsvTest.scala
package top.theonly.spark.sca.sql.textfile

import org.apache.spark.sql.{DataFrame, SparkSession}

/**
 * 读取csv文件
 *  csv文件：由逗号分隔的文件
 */
object SparkReadCsvTest {

  def main(args: Array[String]): Unit = {

    val session: SparkSession = SparkSession.builder()
      .master("local")
      .appName("SparkReadCsvTest")
      .getOrCreate()

//    val frame: DataFrame = session.read
//      // 有列头
//      .option("header", true)
//      .format("csv")
//      .load("./data/sql/test.csv")

    val frame: DataFrame = session.read
      // 有列头
      .option("header", true)
      .csv("./data/sql/test.csv")

    frame.show()
  }

}

File: src/main/java/top/theonly/spark/jav/sql/textfile/SparkReadCsvTest.java
package top.theonly.spark.jav.sql.textfile;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

/**
 * 读取csv文件
 *      用逗号,分隔数据的文件
 */
public class SparkReadCsvTest {
    public static void main(String[] args) {
        SparkSession session = SparkSession.builder().master("local").appName("SparkReadJsonTest").getOrCreate();
        Dataset<Row> ds = session.read()
                // 由列头名
                .option("header", true)
                .csv("./data/sql/test.csv");
        ds.show();

    }
}
